test/draw_line_string.test.js

```diff
@@ -295,52 +276,51 @@ test('draw_line_string interaction', t => {
       // Start a new line
       Draw.deleteAll();
       Draw.changeMode('draw_line_string');
-      click(map, makeMouseEvent(1, 1));
-      click(map, makeMouseEvent(2, 2));
-      click(map, makeMouseEvent(3, 3));
+      mouseClick(map, makeMouseEvent(1, 1));
+      mouseClick(map, makeMouseEvent(2, 2));
+      mouseClick(map, makeMouseEvent(3, 3));
 
       const line = Draw.getAll().features[0];
       st.deepEqual(line.geometry.coordinates, [[1, 1], [2, 2], [3, 3]]);
 
-      const escapeEvent = createSyntheticEvent('keyup', {
-        keyCode: 27
-      });
       container.dispatchEvent(escapeEvent);
 
       st.equal(Draw.getAll().features.length, 0, 'no feature added');
 
-      click(map, makeMouseEvent(1, 1));
+      mouseClick(map, makeMouseEvent(1, 1));
+      map.fire('mousemove', makeMouseEvent(16, 16));
       st.equal(Draw.getAll().features.length, 0, 'no longer drawing');
 
       st.end();
     });
 
+    // ZERO CLICK TESTS
+
     t.test('start a line and end it with Enter', st => {
       // Start a new line
       Draw.deleteAll();
       Draw.changeMode('draw_line_string');
-      click(map, makeMouseEvent(1, 1));
-      click(map, makeMouseEvent(2, 2));
-      click(map, makeMouseEvent(3, 3));
+      mouseClick(map, makeMouseEvent(1, 1));
+      map.fire('mousemove', makeMouseEvent(16, 16));
+      mouseClick(map, makeMouseEvent(2, 2));
+      mouseClick(map, makeMouseEvent(3, 3));
 
       const line = Draw.getAll().features[0];
       st.deepEqual(line.geometry.coordinates, [[1, 1], [2, 2], [3, 3]]);
 
-      const enterEvent = createSyntheticEvent('keyup', {
-        keyCode: 13
-      });
       container.dispatchEvent(enterEvent);
 
       st.equal(Draw.getAll().features.length, 1, 'the feature was added');
       st.deepEqual(Draw.getAll().features[0].geometry.coordinates, [[1, 1], [2, 2], [3, 3]], 'the line is correct');
 
-      click(map, makeMouseEvent(1, 1));
+      mouseClick(map, makeMouseEvent(1, 1));
+      map.fire('mousemove', makeMouseEvent(16, 16));
       st.equal(Draw.getAll().features.length, 1, 'no longer drawing');
 
       st.end();
     });
 
-    t.test('start draw_line_string mode then exit before a click', st => {
+    t.test('start draw_line_string mode then change_mode before a click', st => {
```

Comment:
change mode without underscore, right? underscore makes it seem like it's a mode